x/programs/runtime/storage.go

```diff
@@ -0,0 +1,131 @@
+package runtime
+
+import (
+	"encoding/binary"
+	"errors"
+
+	"github.com/ava-labs/avalanchego/database"
+	"github.com/ava-labs/hypersdk/consts"
+	"github.com/ava-labs/hypersdk/crypto/ed25519"
+)
+
+const (
+	HRP                = "simulator_key_"
+	programPrefix      = 0x0
+	programCountPrefix = 0x1
+	storagePrefix      = 0x2
+)
+
+func ProgramKey(asset uint64) (k []byte) {
```

Comment:
I think the VM will probably want to define their own storage logic. Which is why I didn't go this route to begin with and was poking around the storage interface. Otherwise each VM will be implementing the runtime interface which I don't think we want. They **could** but I would imagine most to use the implementation in hypersdk when it becomes stable. So this needs more thought.

Reply:
per conversation offline we can change this after Friday.